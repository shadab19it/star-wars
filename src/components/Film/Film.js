import React from "react";
import { Row } from "react-bootstrap";
import PropTypes from "prop-types";
import _ from "lodash";
import ButtonStandard from "../ButtonStandard/ButtonStandard";
import "./Film.scss";
import moment from "moment";

function Film(props) {
    const { data, editFilm } = props;

    const lodashTruncate = (data, maxLength) => {
        return _.truncate(data, {
            length: maxLength
        });
    };

    if (data !== undefined && typeof data == "object") {
        const releaseDate = moment(data.release_date).format("DD/MM/YYYY");

        const newData = {
            title: data.title,
            director: data.director,
            releaseDate: releaseDate
        };

        let cols = 12;
        const objects = Object.entries(newData);

        const math = Number((cols / objects.length).toFixed(1));

        const size = math > 1 ? math : 1;

        const maxLength = size * 12;

        const colEdit = (
            <div className="col-3 custom-col">
                <div className="custom-checkbox">
                    <ButtonStandard
                        text="Detalhes"
                        textClass="text-2"
                        className="btn btn-primary btn-rounded btn-standard px-4 ml-3"
                        onClick={() => editFilm(data.episode_id)}
                    />
                </div>
            </div>
        );

        if (objects.length > 0) {
            return (
                <Row className="custom-film-row my-2">
                    {objects.map((data, i) => {
                        return (
                            <div className="col-3" key={i}>
                                <span className="text-2 font-weight-bolder">
                                    {lodashTruncate(data[1], maxLength)}
                                </span>
                            </div>
                        );
                    })}
                    {colEdit}
                </Row>
            );
        } else {
            return <p>Nenhum filme encontrado.</p>;
        }
    } else {
        return <p>Dados inv√°lidos</p>;
    }
}

Film.propTypes = {
    data: PropTypes.object.isRequired,
    id: PropTypes.string
};
export default Film;
