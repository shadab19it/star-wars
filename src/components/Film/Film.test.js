import React from "react";
import Film from "./Film";
import { IntlProvider } from "react-intl";
import { configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

const country = "pt-BR";

const filmData = {
    episode_id: "1",
    title: "A New Hope",
    release_date: "1977-05-25"
};

describe("Film Component", () => {
    const component = (
        <IntlProvider locale={country}>
            <Film data={filmData} functionProps={() => true} id={filmData.id} />
        </IntlProvider>
    );

    it("matches the snapshot", () => {
        const tree = mount(component);
        expect(tree).toMatchSnapshot();
    });

    it("generated a film item with the details button", () => {
        const wrapper = mount(component);
        const button = wrapper.find("button");
        expect(button.length).toEqual(1);
    });

    it("generated a film item with the columns showing", () => {
        const wrapper = mount(component);
        const checkbox = wrapper
            .find(".text-2")
            .first()
            .text();
        expect(checkbox).toBe("A New Hope");
    });
});
