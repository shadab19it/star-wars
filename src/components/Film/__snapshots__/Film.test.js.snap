// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film Component matches the snapshot 1`] = `ReactWrapper {}`;
